### PR TITLE
Tzachi libre2 support 1

### PR DESCRIPTION
**Pull request by tzachi-dar**
Code that can connect to libre2 and receive data using BT.

On the first time that the user will scan the sensor, he will get a message asking weather to start the new sensor.
It replaces https://github.com/NightscoutFoundation/xDrip/pull/1429

Ready for code review.

### CODE REVIEW

**Comment by keencave** [comment]
Tested both OOP2 and direct Libre2 connect successful for few weeks now. Both modes are working stable and reliable. I strongly recommend to include this PR soon as it enables full calibration mode for Libre2. Users still reports intercepts of > 20 mg/dl for Libre2 which would be fixed then.

**Review by chrisdex — APPROVED** [review]

**Review by chrisdex — APPROVED** [review]

File: app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java [hunk 1]
```diff
@@ -94,7 +95,7 @@
 public class DexCollectionService extends Service implements BtCallBack {
     public static final String LIMITTER_NAME = "LimiTTer";
     private final static String TAG = DexCollectionService.class.getSimpleName();
-    private static final boolean d = false;
+    private static final boolean d = true;
```
Comment (jamorham) [inline app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java:98]: Does this need to be true?

**Review by jamorham — APPROVED** [review]
This looks good to me, I don't fully follow all the changes to the flow but I understand it has been tested and is ready for being tested more widely

**Comment by Electrik-rich546456** [comment]
Hello @keencave Ive been using xdrip to receive the data from my FSL2 sensor for a long time now and always thought it was silly to have the need of OOP2 to decode the data from the sensor. Considering this [app](https://github.com/j-kaltes/Juggluco) can do it, why can't xdrip? 
Am i mistaken by the looks of your comment [on Dec 3, 2020](https://github.com/NightscoutFoundation/xDrip/pull/1524#issuecomment-737964226) .

Does anyone know if there is a branch that has the ability so send the data gathered to libreview for my health provider to see my levels and cards and insulin administered

**Pull request merged**